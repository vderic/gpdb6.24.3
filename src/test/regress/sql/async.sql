--
-- ASYNC
--

--Should work. Send a valid message via a valid channel name
SELECT pg_notify('notify_async1','sample message1');
SELECT pg_notify('notify_async1','');
SELECT pg_notify('notify_async1',NULL);

-- Should fail. Send a valid message via an invalid channel name
SELECT pg_notify('','sample message1');
SELECT pg_notify(NULL,'sample message1');
SELECT pg_notify('notify_async_channel_name_too_long______________________________','sample_message1');

--Should work. Valid NOTIFY/LISTEN/UNLISTEN commands
NOTIFY notify_async2;
LISTEN notify_async2;
UNLISTEN notify_async2;
UNLISTEN *;

-- start_matchsubs
-- m/Asynchronous notification "notify_async3" received from server process with PID.*/
-- s/Asynchronous notification "notify_async3" received from server process with PID.*/Asynchronous notification "notify_async3" received/
-- end_matchsubs
\c postgres
LISTEN notify_async3;
\! psql postgres -c "notify notify_async3;"
SELECT;
\c -
