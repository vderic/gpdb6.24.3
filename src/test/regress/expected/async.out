--
-- ASYNC
--
--Should work. Send a valid message via a valid channel name
SELECT pg_notify('notify_async1','sample message1');
 pg_notify 
-----------
 
(1 row)

SELECT pg_notify('notify_async1','');
 pg_notify 
-----------
 
(1 row)

SELECT pg_notify('notify_async1',NULL);
 pg_notify 
-----------
 
(1 row)

-- Should fail. Send a valid message via an invalid channel name
SELECT pg_notify('','sample message1');
ERROR:  channel name cannot be empty
SELECT pg_notify(NULL,'sample message1');
ERROR:  channel name cannot be empty
SELECT pg_notify('notify_async_channel_name_too_long______________________________','sample_message1');
ERROR:  channel name too long
--Should work. Valid NOTIFY/LISTEN/UNLISTEN commands
NOTIFY notify_async2;
LISTEN notify_async2;
UNLISTEN notify_async2;
UNLISTEN *;
-- start_matchsubs
-- m/Asynchronous notification "notify_async3" received from server process with PID.*/
-- s/Asynchronous notification "notify_async3" received from server process with PID.*/Asynchronous notification "notify_async3" received/
-- end_matchsubs
\c postgres
LISTEN notify_async3;
\! psql postgres -c "notify notify_async3;"
NOTIFY
SELECT;
--
(1 row)

Asynchronous notification "notify_async3" received
\c -
